src/formatter-settings/assets/features/formatterSettings/formatterSettingViewSlice.ts

```diff
@@ -39,26 +40,39 @@ export const formatterSettingsViewSlice = createSlice({
           state.exampleKind = ExampleKind.WRAPPING_EXAMPLE;
           break;
       }
-      state.formattedContent = Example.getExample(state.exampleKind);
+      onWillChangeExampleKind(state.exampleKind);
     },
-    changeSetting: (state, action) => {
-      for (const setting of state.settings) {
-        if (setting.id === action.payload.id) {
-          setting.value = action.payload.value;
-          if (setting.exampleKind !== state.exampleKind) {
-            state.exampleKind = setting.exampleKind;
-            state.formattedContent = Example.getExample(state.exampleKind);
-          }
-          break;
+    loadProfileSetting: (state, action) => {
+      state.profileSettings = action.payload.setting;
+    },
+    loadVSCodeSetting: (state, action) => {
+      state.vscodeSettings = action.payload.setting;
+      for (const setting of state.vscodeSettings) {
+        if (setting.id === SupportedSettings.TABULATION_SIZE) {
+          document.documentElement.style.setProperty("--vscode-tab-size", setting.value);
+        } else if (setting.id === VSCodeSettings.DETECT_INDENTATION) {
+          state.detectIndentation = (setting.value === "true");
         }
       }
     },
+    activateExampleKind: (state, action) => {
```

Comment:
This reducer is wired. Why sending a request to the server in a reducer?

Reply:
Will remove requests in the reducer.